core/src/main/java/kafka/server/RemoteClusterMetadataManager.java

```diff
@@ -212,7 +211,68 @@ public void refreshRemoteMetadata() {
+                    topic -> new DescribeConfigsRequestData.DescribeConfigsResource().
+                    setResourceType(ConfigResource.Type.TOPIC.id()).setResourceName(topic)).toList();
+            DescribeConfigsRequest.Builder describeConfigsRequest = new DescribeConfigsRequest.Builder(new DescribeConfigsRequestData().setResources(describeConfigsResources));
+
+            Map<String, Map<String, String>> configsToChange = new HashMap<>();
+            Map<String, String> conChange = new HashMap<>();
+            var describeConfigResponse = senders.get(random.nextInt(senders.size())).sendRequest(describeConfigsRequest);
+            if (describeConfigResponse.responseBody() instanceof DescribeConfigsResponse describeConfigsRes) {
+                log.debug("!!! periodic describeConfigsRes: {}", describeConfigsRes);
+                describeConfigsRes.data().results().forEach(describeConfigResult -> {
+                    if (describeConfigResult.resourceType() == ConfigResource.Type.TOPIC.id() && topics.get(clusterLinkName).contains(describeConfigResult.resourceName())) {
+                        Properties props = metadataCache.topicConfig(describeConfigResult.resourceName());
+                        describeConfigResult.configs().forEach(con -> {
+                            if (con.configSource() == DescribeConfigsResponse.ConfigSource.TOPIC_CONFIG.id()) {
+                                if (props.containsKey(con.name())) {
+                                    if (!props.get(con.name()).equals(con.value())) {
+                                        conChange.put(con.name(), con.value());
+                                    }
+                                } else {
+                                    conChange.put(con.name(), con.value());
+                                }
+
+                            }
+                        });
+                        if (!conChange.isEmpty()) {
+                            configsToChange.put(describeConfigResult.resourceName(), new HashMap<>(conChange));
+                            conChange.clear();
+                        }
+
+                    }
+
+                });
+            }
+
+            log.info("!!! applying config change: {}", configsToChange);
+            // apply the change
+            Map<ConfigResource, Collection<AlterConfigOp>> configOps = new HashMap<>();
+            configsToChange.forEach((name, changes) -> {
+                var changeList = changes.entrySet().stream().map(entry -> new AlterConfigOp(new ConfigEntry(entry.getKey(), entry.getValue()),  AlterConfigOp.OpType.SET)).toList();
+                configOps.put(new ConfigResource(ConfigResource.Type.TOPIC, name), changeList);
+            });
+
+            if (!configOps.isEmpty()) {
+                try {
+                    var result = adminClient.incrementalAlterConfigs(configOps).all().get(60, TimeUnit.SECONDS);
```

Comment:
I should use `channelManager` here, instead of adminClient. Will update tomorrow.

Reply:
PR Updated!